core/src/main/java/org/apache/iceberg/PartitionsTable.java

```diff
@@ -167,16 +172,25 @@ Partition get(StructLike key) {
     private final StructLike key;
     private long recordCount;
     private int fileCount;
+    private int deleteFileCount;
 
     Partition(StructLike key) {
       this.key = key;
       this.recordCount = 0;
       this.fileCount = 0;
```

Comment:
If we're going to add this, likely we should also make this `dataFileCount`.

But I would hold off on updating that part, as this is a somewhat larger change to the spec and we should ensure that it's going to be considered in its current form before making smaller updates like that (so you don't spend time unnecessarily).

Reply:
Thanks for your attention, but I think this change is necessary. In our company, users always query the partitions to check if the table is well compaction. When partial compaction is enabled, there may be some partition fails to compaction, but without this query, we can not know which partition is bad.

Reply:
Yea, this does seem like it should be renamed "file_count" to "data_file_count" if we can...(not sure the policy here), agree the original name is unfortunate.